storm-core/src/jvm/org/apache/storm/container/cgroup/core/CpusetCore.java

```diff
@@ -0,0 +1,209 @@
+    }
+
+    public boolean isMemHardwall() throws IOException {
+        int output = Integer.parseInt(CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CPUSET_MEM_HARDWALL)).get(0));
+        return output > 0;
+    }
+
+    public int getMemPressure() throws IOException {
+        String output = CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CPUSET_MEMORY_PRESSURE)).get(0);
+        return Integer.parseInt(output);
+    }
+
+    public void setMemPressureEnabled(boolean flag) throws IOException {
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, CPUSET_MEMORY_PRESSURE_ENABLED), String.valueOf(flag ? 1 : 0));
+    }
+
+    public boolean isMemPressureEnabled() throws IOException {
+        int output = Integer.parseInt(CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CPUSET_MEMORY_PRESSURE_ENABLED)).get(0));
+        return output > 0;
+    }
+
+    public void setMemSpreadPage(boolean flag) throws IOException {
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, CPUSET_MEMORY_SPREAD_PAGE), String.valueOf(flag ? 1 : 0));
+    }
+
+    public boolean isMemSpreadPage() throws IOException {
+        int output = Integer.parseInt(CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CPUSET_MEMORY_SPREAD_PAGE)).get(0));
+        return output > 0;
+    }
+
+    public void setMemSpreadSlab(boolean flag) throws IOException {
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, CPUSET_MEMORY_SPREAD_SLAB), String.valueOf(flag ? 1 : 0));
+    }
+
+    public boolean isMemSpreadSlab() throws IOException {
+        int output = Integer.parseInt(CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CPUSET_MEMORY_SPREAD_SLAB)).get(0));
+        return output > 0;
+    }
+
+    public void setSchedLoadBlance(boolean flag) throws IOException {
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, CPUSET_SCHED_LOAD_BALANCE), String.valueOf(flag ? 1 : 0));
+    }
+
+    public boolean isSchedLoadBlance() throws IOException {
+        int output = Integer.parseInt(CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CPUSET_SCHED_LOAD_BALANCE)).get(0));
+        return output > 0;
+    }
+
+    public void setSchedRelaxDomainLevel(int value) throws IOException {
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, CPUSET_SCHED_RELAX_DOMAIN_LEVEL), String.valueOf(value));
+    }
+
+    public int getSchedRelaxDomainLevel() throws IOException {
+        String output = CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CPUSET_SCHED_RELAX_DOMAIN_LEVEL)).get(0);
+        return Integer.parseInt(output);
+    }
+
+    public static int[] parseNums(String outputStr) {
+        char[] output = outputStr.toCharArray();
+        LinkedList<Integer> numList = new LinkedList<Integer>();
+        int value = 0;
+        int start = 0;
+        boolean isHyphen = false;
+        for (char ch : output) {
```

Comment:
I agree this is not the most straightforward way of doing this in Java and
looks more C-like. 

I think I would be OK with leaving the code as-is, and maybe we could add a
comment explaining what this does for others since it is? We could clean it up
in follow-on work if it still is a problem.

```
/** 
 * Accepts a string of comma-separated values or ranges of values.  Specify a
 * value like '42' and a range like '12-17'.
 *
 * Example: '42,12-17,100' -> {42, 12, 13, 14, 15, 16, 17, 100}
 */
```